app/index.js

```diff
@@ -190,14 +194,35 @@ module.exports = yeoman.Base.extend({
     },
 
     composerJson: function () {
-      var composer = this.fs.readJSON('composer.json');
+      // Import any existing composer.json that was saved above.
+      var composer = {};
+      var composerExists = this.fs.exists('composer.orig');
+      // Project hasn't been generated yet, so start with composer template.
+      // This was copied from the gdt template in gdtBase() above.
+      if (composerExists) {
+        composer = this.fs.readJSON('composer.orig');
+      }
+      else {
+        composer = this.fs.readJSON('composer.json');
+      }
+
       composer.name = options.projectName;
       composer.description = options.projectDescription;
+      // Allow distros to modify the composer.json.
+      if (typeof options.drupalDistro.modifyComposer == 'function') {
+        var done = this.async();
+        composer = options.drupalDistro.modifyComposer(this, options, composer, composerExists, done);
```

Comment:
Do we think different distros that want to modify the composer.json surgically like this will want to modify it in different ways?